### PR TITLE
PCHR-1477: Remove Cases tab if T&A is enabled

### PR DESCRIPTION
**Pull request by AkA84**
**Before**
before

**After**
after

Also refactored the code a bit as well

### CODE REVIEW

File: hrui/hrui.php [hunk 1]
```diff
@@ -33,13 +33,19 @@ function hrui_civicrm_pageRun($page) {
...
+
   if ($page instanceof CRM_Contact_Page_View_Summary) {
     CRM_Core_Resources::singleton()
       ->addScriptFile('org.civicrm.hrui', 'js/contact.js')
       ->addScriptFile('org.civicrm.hrui', 'js/hrui.js')
-      ->addSetting(array('pageName' => 'viewSummary'));
+      ->addSetting(array('pageName' => 'viewSummary'))
+      ->addSetting(array('extensionsEnabled' => array(
+        'taskAssignments' => !!CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Extension', 'uk.co.compucorp.civicrm.tasksassignments', 'is_active', 'full_name')
+      )));
```
Comment (omarabuhussein) [inline hrui/hrui.php]: you can wrap checking for T&A if enabled or not in a seperate method , something like what i did here for hrcase https://github.com/civicrm/civihr/blob/ab6a7b72984a481717f2535bbeaf90f0dce98ea1/hrcase/hrcase.php#L506

also it's better not to do my mistake and make the method receive any extension name ( for more usability in the feature )

**Comment by omarabuhussein** [comment]
I believe there is a better way to do this rather than using javascript to achieve that ... there is this hook which you can use to do that , there is even an example to remove the contribition tab https://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_tabset

File: hrui/hrui.php [hunk 2]
```diff
@@ -399,24 +396,44 @@ function hrui_civicrm_upgrade($op, CRM_Queue_Queue $queue = NULL) {
...
 
 /**
- * Implementation of hook_civicrm_tabs
+ * Check if the extension with the given key is enabled
+ *
+ * @param string $extensionKey
+ * @return boolean
+ */
+function _hrui_check_extension($extensionKey)  {
+  return !!CRM_Core_DAO::getFieldValue(
```
Comment (omarabuhussein) [inline hrui/hrui.php]: you can cast the value to boolean :

return (boolean) CRM_Core_DAO::getFieldValue ... etc 

it's more clear like that.

**Comment by omarabuhussein** [comment]
just one minor note which is up to you to do or not , otherwise approved

**Comment by jamienovick** [comment]
We should definitely do this via hook not via javascript please. Can we make sure that we are using the hooks as much as possible? J

**Comment by AkA84** [comment]
@jamienovick it was indeed done via hook after @omarabuhussein 's comment

**Pull request merged**